### PR TITLE
Added default go version to 1.17

### PR DESCRIPTION
**Pull request by evalsocket**
# TL;DR
Added go-version in reuse able workflow

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
_Remove the '*fixes*' keyword if there will be multiple PRs to fix the linked issue_

fixes https://github.com/flyteorg/flyte/issues/<number>

## Follow-up issue
_NA_
OR
_https://github.com/flyteorg/flyte/issues/<number>_

### CODE REVIEW

**Comment by pingsutw** [comment]
awesome, thank you @evalsocket

**Review by pingsutw — APPROVED** [review]

**Review by katrogan — APPROVED** [review]
